docs/core/tools/global-tools-how-to-create.md

```diff
@@ -0,0 +1,226 @@
+---
+title: How to create a .NET Core Global Tool
+description: Describes how to create a Global Tool. The Global Tool is a console application that is installed through the .NET Core CLI.
+author: Thraka
+ms.author: adegeo
+ms.date: 08/22/2018
+---
+
+# .NET Core Global Tools overview
+
+This article teaches you how to create and package a .NET Core Global Tool. The .NET Core CLI allows you to compile and mark a console application as a Global Tool, which others can easily install. .NET Core Global Tools are NuGet packages that are installed from the .NET Core CLI. For more information about Global Tools, see [.NET Core Global Tools overview][global-tool-info].
+
+[!INCLUDE [topic-appliesto-net-core-21plus.md](../../../includes/topic-appliesto-net-core-21plus.md)]
+
+## Create a project
+
+This article shows uses the .NET Core CLI to create and manage a project. Obviously you could use Visual Studio to do some of these steps, but Visual Studio will not be described in this article. 
+
+First, create a new .NET Core 2.1 Console Application.
+
+```console
+dotnet new console -o botsay
+```
+
+Enter the `botsay` directory created by the previous command.
+
+In this example we'll create a console application that generates an ascii bot and prints a message. The first thing we'll do is use the same command line parsing system that the .NET Core CLI tool uses.
+
+Add reference to the `Microsoft.Extensions.CommandLineUtils` NuGet package.
+
```

Comment:
So, this is complicated...

You have latched on to exactly the big problem in creating a global tool, but .NET doesn't provide a decent answer. 

Microsoft.Extensions.CommandLineUtils is not under active development, so we don't really encourage folks to use it, although it is likely to stay in its current configuration on NuGet. So technically, this isn't wrong, but we should consider carefully and at the least not make much of a recommendation for it. 

It is wrong that it is what .NET Core CLI uses. There is an interesting lineage, and the CLI parser is in the same lineage, but the CLI carries it's own parsing code. 

The two suggestions I make are to use Nate McMasters fork of Microsoft.Extensions.CommandLineUtils at https://github.com/natemcmaster/CommandLineUtils or to wait for System.CommandLine which is a side project a bunch of people including those involved in the previous MS lineage command line parsers are working on. 

For this purpose, I think we should have two articles: One similar to Rich's dotnetsay that simply grabs the argument from the array and hardcodes help. I think this is actually a good idea on something as simple as this code. And if we simplify it, consider naming dotnetsay as it gives better credit to the bot :)

There is useful code for people here, and I make more comments on that.

Reply:
I did find out that the CommandLineUtils was no longer being developed, and that natemcmaster did have a version up there, but I wasn't sure if I could use it.